### PR TITLE
HW USSD msisdn update check functionality

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**
The message receiver's phone number should be checked against the contacts DB and contact details updated if necessary.  This check and update functionality should be added to state 2

### CODE REVIEW

**Comment by NicolaasDieBaas** [comment]
@gsvr please review.

File: test/ussd_health_worker.test.js [hunk 1]
```diff
@@ -160,6 +164,40 @@ describe("familyconnect health worker app", function() {
...
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '12345'  // state_auth_code - personnel code
+                        , '1'  // state_msg_receiver - head of household
+                        , '08200000444'  // state_msisdn
+                    )
+                    .check.interaction({
+                        state: 'state_msisdn_already_registered',
+                        reply: [
+                            "[MSISDN] is already registered for messages.",
```
Comment (gsvr) [inline test/ussd_health_worker.test.js]: replace [MSISDN] with the actual number

File: test/ussd_health_worker.test.js [hunk 2]
```diff
@@ -160,6 +164,40 @@ describe("familyconnect health worker app", function() {
...
                     .run();
             });
+            it("to state_msisdn_already_registered", function() {
+                return tester
+                    .setup.user.addr('082111')
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '12345'  // state_auth_code - personnel code
+                        , '1'  // state_msg_receiver - head of household
+                        , '08200000444'  // state_msisdn
```
Comment (gsvr) [inline test/ussd_health_worker.test.js]: Uganda numbers are 10 digit long, so let's use 10-digit numbers.

File: test/fixtures.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 // Contact roles
 // 082111: registered health worker
 // 082222: unregistered person
+// 08200000333: unregistered person (with valid msidn)
```
Comment (gsvr) [inline test/fixtures.js]: msisdn

File: test/ussd_health_worker.test.js [hunk 4]
```diff
@@ -160,6 +164,40 @@ describe("familyconnect health worker app", function() {
...
+                        , '1'  // state_msg_receiver - head of household
+                        , '08200000555'  // state_msisdn
+                    )
+                    .check.interaction({
+                        state: 'state_household_head_name',
+                        reply: "Please enter the first name of the Head of the Household. For example: Isaac."
+                    })
+                    .run();
+            });
             it("to state_household_head_name", function() {
```
Comment (gsvr) [inline test/ussd_health_worker.test.js]: Let's add a description in brackets like `(unregistered msisdn)`

File: test/ussd_health_worker.test.js [hunk 5]
```diff
@@ -528,7 +566,7 @@ describe("familyconnect health worker app", function() {
                         {session_event: 'new'}  // dial in
                         , '12345'  // state_auth_code - personnel code
                         , '2'  // state_msg_receiver - mother to be
-                        , '0803304899'  // state_msisdn
+                        , '08200000333'  // state_msisdn
```
Comment (gsvr) [inline test/ussd_health_worker.test.js]: 10-digit

File: test/ussd_health_worker.test.js [hunk 6]
```diff
@@ -160,6 +164,40 @@ describe("familyconnect health worker app", function() {
...
+                    .run();
+            });
+            it("to state_household_head_name (registered msisdn with no subscriptions)", function() {
+                return tester
+                    .setup.user.addr('082111')
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '12345'  // state_auth_code - personnel code
+                        , '1'  // state_msg_receiver - head of household
+                        , '08200000555'  // state_msisdn
```
Comment (gsvr) [inline test/ussd_health_worker.test.js]: 10-digit

**Comment by NicolaasDieBaas** [comment]
@gsvr 2nd pass review plz

**Comment by gsvr** [comment]
:+1:

**Pull request merged**